### PR TITLE
fix: [CI-4356]: added feature for fetching repositories

### PR DESCRIPTION
**Pull request by aman-harness**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32862)

### CODE REVIEW

**Comment by aman-harness** [comment]
trigger messagecheck

File: 120-ng-manager/src/main/java/io/harness/ng/trialsignup/ProvisionService.java [hunk 1]
```diff
@@ -267,4 +275,22 @@ public DelegateStatus getDelegateInstallStatus(String accountId) {
...
     }
   }
+
+  public List<UserRepoResponse> getAllUserRepos(String accountId, String repoRef) {
+    Optional<ConnectorResponseDTO> connector = connectorService.getByRef(accountId, null, null, repoRef);
+    connector.orElseThrow(
+        () -> new InvalidArgumentsException(format("connector %s was not found in account %s", repoRef, accountId)));
+    ScmConnector decryptedConnector = gitSyncConnectorHelper.getDecryptedConnector(
+        accountId, null, null, (ScmConnector) connector.get().getConnector().getConnectorConfig());
+    return convertToUserRepo(scmClient.getAllUserRepos(decryptedConnector));
```
Comment (HenAmar) [inline 120-ng-manager/src/main/java/io/harness/ng/trialsignup/ProvisionService.java:285]: We need to update the scm object to be repo instead of account.

**Review by HenAmar — CHANGES_REQUESTED** [review]

**Review by HenAmar — APPROVED** [review]
> Reviewed 2 of 10 files at r1, 10 of 12 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32862)* status: :shipit: complete! ✓

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 2 of 10 files at r1, 10 of 12 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32862)* status: :shipit: complete! ✓

**Pull request merged**